Tests/MongoSwiftSyncTests/SpecTestRunner/CodableExtensions.swift

```diff
@@ -64,33 +64,54 @@ extension TransactionOptions: Decodable {
     }
 }
 
-extension ReadPreference.Mode: Decodable {}
+extension ReadPreference.Mode: Decodable {
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.singleValueContainer()
+        var string = try container.decode(String.self)
+        // spec tests capitalize first letter of mode, so need to account for that.
+        string = string.prefix(1).lowercased() + string.dropFirst()
+        guard let mode = Self(rawValue: string) else {
+            throw DecodingError.dataCorruptedError(
+                in: container,
+                debugDescription: "can't parse ReadPreference mode from \(string)"
+            )
+        }
+        self = mode
+    }
+}
 
 extension ReadPreference: Decodable {
     private enum CodingKeys: String, CodingKey {
         case mode
     }
 
     public init(from decoder: Decoder) throws {
-        let container = try decoder.container(keyedBy: CodingKeys.self)
-        let mode = try container.decode(Mode.self, forKey: .mode)
-        self.init(mode)
+        if let container = try? decoder.container(keyedBy: CodingKeys.self) {
+            let mode = try container.decode(Mode.self, forKey: .mode)
+            self.init(mode)
+        } else { // sometimes the spec tests only specify the mode as a string
+            let container = try decoder.singleValueContainer()
+            let mode = try container.decode(ReadPreference.Mode.self)
+            self.init(mode)
+        }
     }
 }
 
 extension MongoClientOptions: Decodable {
     private enum CodingKeys: String, CodingKey {
-        case retryReads, retryWrites, w, readConcernLevel, mode = "readPreference"
+        case retryReads, retryWrites, w, readConcernLevel, readPreference, heartbeatFrequencyMS
```

Comment:
heartbeatFrequencyMS is specified in some of the transactions tests and we weren't picking it up. Once I started lowering it all the way to 50, some tests started to fail. Now it only lowers to 50 if the spec tests don't specify a heartbeatFrequencyMS.